modules/renter/contractor/persist.go

```diff
@@ -53,7 +53,18 @@ func (c *Contractor) load() error {
 	for _, contract := range data.Contracts {
 		c.contracts[contract.ID] = contract
 	}
-	c.financialMetrics = data.FinancialMetrics
+	c.currentPeriod = data.CurrentPeriod
+	if c.currentPeriod == 0 {
+		// If loading old persist, current period will be unknown. Best we can
+		// do is guess based on contracts + allowance.
```

Comment:
IIRC our current versioning scheme is to only bump the persist version when the data is no longer convertible. The persist package takes the simple approach of checking for basic version equality, rather than a more nuanced comparison (e.g `>=v1.0.4`).

When it comes time to write the compatibility code, we should probably switch to using `build.Version` instead of individual persist versions. Then the compat code can define ranges of versions that are compatible.